checks.d/file.py

```diff
@@ -0,0 +1,85 @@
+import errno
+import os
+import time
+
+from checks import AgentCheck
+
+class FileCheck(AgentCheck):
+
+    STATUS_ABSENT = 'absent'
+    STATUS_PRESENT = 'present'
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self._last_state_by_path = {}
+
+    def has_different_status(self, path, current):
+        last_state = self._last_state_by_path.get(path, None)
+        self._last_state_by_path[path] = current
+        return (last_state is not None and last_state != current)
+
+    def stat_file(self, path):
+        try:
+            statinfo = os.stat(path)
+            return self.STATUS_PRESENT, statinfo
+        except OSError, e:
+            if e.errno == errno.ENOENT:
+                return self.STATUS_ABSENT, []
+            else:
+                raise
+
+    def check(self, instance):
+        """
+        Stats a file and emits service_checks and metrics on file creation/age.
+        """
+        if 'path' not in instance:
+            raise Exception("Missing 'path' in file check config")
+
+        path = instance['path']
+        expect = instance['expect']
+
+        status, statinfo = self.stat_file(path)
+
+        # Emit a service check:
+        msg = "File %s is %s" % (path, expect)
+        check_status = AgentCheck.OK
+        service_tags = [
```

Comment:
Breaking out the common tags here explicitly makes it a bit easier for me to read, e.g.:
```python
base_tags = [
  'expected_status:' + expect,
  'path:' + path,
  'actual_status': + status,
]
⋮
service_tags = base_tags + []
⋮
age_tags = base_tags + []
```